services/cd-service/pkg/repository/transformer.go

```diff
@@ -3754,22 +3754,23 @@ func (c *envReleaseTrain) Transform(
 			}
 			releaseDir := releasesDirectoryWithVersion(state.Filesystem, appName, release)
 			newEvent := createLockPreventedDeploymentEvent(appName, c.Env, prognosis.FirstLockMessage, "environment")
-
-			if state.DBHandler.ShouldUseOtherTables() {
-				commitID, err := getCommitID(ctx, transaction, state, state.Filesystem, release, releaseDir, appName)
-				if err != nil {
-					logger.FromContext(ctx).Sugar().Warnf("could not write event data - continuing. %v", fmt.Errorf("getCommitIDFromReleaseDir %v", err))
-				} else {
-					gen := getGenerator(ctx)
-					eventUuid := gen.Generate()
-					err = state.DBHandler.DBWriteLockPreventedDeploymentEvent(ctx, transaction, c.TransformerEslVersion, eventUuid, commitID, newEvent)
+			if c.WriteCommitData {
+				if state.DBHandler.ShouldUseOtherTables() {
+					commitID, err := getCommitID(ctx, transaction, state, state.Filesystem, release, releaseDir, appName)
 					if err != nil {
```

Comment:
if we get three levels deep in conditionals, we should consider to refactor out into a separate function (also for enabling unittesting with decent coverage).

Reply:
I refactored it so that we don't have three-level deep conditionals.
Now we just return before the parsing logic if `c.WriteCommitData` is not enabled.